processing/src/main/java/io/druid/query/aggregation/last/DoubleLastAggregatorFactory.java

```diff
@@ -0,0 +1,250 @@
+  }
+
+  @Override
+  public AggregatorFactory getCombiningFactory()
+  {
+    return new DoubleLastAggregatorFactory(name, name)
+    {
+      @Override
+      public Aggregator factorize(ColumnSelectorFactory metricFactory)
+      {
+        final ObjectColumnSelector selector = metricFactory.makeObjectColumnSelector(name);
+        return new DoubleLastAggregator(name, null, null)
+        {
+          @Override
+          public void aggregate()
+          {
+            SerializablePair<Long, Long> pair = (SerializablePair<Long, Long>) selector.get();
+            if (pair.rhs >= lastTime) {
+              lastTime = pair.lhs;
+              lastValue = pair.rhs;
+            }
+          }
+        };
+      }
+
+      @Override
+      public BufferAggregator factorizeBuffered(ColumnSelectorFactory metricFactory)
+      {
+        final ObjectColumnSelector selector = metricFactory.makeObjectColumnSelector(name);
+        return new DoubleLastBufferAggregator(null, null)
+        {
+          @Override
+          public void aggregate(ByteBuffer buf, int position)
+          {
+            SerializablePair<Long, Double> pair = (SerializablePair<Long, Double>) selector.get();
+            long lastTime = buf.getLong(position);
+            if (pair.lhs >= lastTime) {
+              buf.putLong(position, pair.lhs);
+              buf.putDouble(position + Longs.BYTES, pair.rhs);
+            }
+          }
+        };
+      }
+    };
+  }
+
+  @Override
+  public AggregatorFactory getMergingFactory(AggregatorFactory other) throws AggregatorFactoryNotMergeableException
+  {
+    if (other.getName().equals(this.getName()) && this.getClass() == other.getClass()) {
+      return getCombiningFactory();
+    } else {
+      throw new AggregatorFactoryNotMergeableException(this, other);
+    }
+  }
+
+  @Override
+  public List<AggregatorFactory> getRequiredColumns()
+  {
+    return Arrays.<AggregatorFactory>asList(new LongFirstAggregatorFactory(fieldName, fieldName));
+  }
+
+  @Override
+  public Object deserialize(Object object)
+  {
+    Map map = (Map) object;
+    return new SerializablePair<>(((Number) map.get("lhs")).longValue(), ((Number) map.get("rhs")).doubleValue());
+  }
+
+  @Override
+  public Object finalizeComputation(Object object)
+  {
+    return ((SerializablePair<Long, Double>) object).rhs;
+  }
+
+  @Override
+  @JsonProperty
+  public String getName()
+  {
+    return name;
+  }
+
+  @JsonProperty
+  public String getFieldName()
+  {
+    return fieldName;
+  }
+
+  @Override
+  public List<String> requiredFields()
+  {
+    return Arrays.asList(Column.TIME_COLUMN_NAME, fieldName);
+  }
+
+  @Override
+  public byte[] getCacheKey()
+  {
+    byte[] fieldNameBytes = StringUtils.toUtf8(fieldName);
+
+    return ByteBuffer.allocate(1 + fieldNameBytes.length).put(CACHE_TYPE_ID).put(fieldNameBytes).array();
+  }
+
+  @Override
+  public String getTypeName()
+  {
+    return "float";
```

Comment:
Sorry if I misunderstood, but are you suggesting to have getTypeName return some type other than float or long?